### DISCUSSION TITLE
[WASM] Optimizing AOT startup behavior

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#13063] [Q&A]
I'm looking into optimizing the startup behavior of my UNO WASM app. 

Here is my setup:
- hosted on a AWS S3 bucket
- UNO WASM is full AOT - I don't want to look into profile guided AOT at this point

Findings:
- browser caching definitely works. However, the browser sends out [conditional get](https://developer.mozilla.org/en-US/docs/Web/HTTP/Conditional_requests) requests on every restart - or even on F5 when waiting for while before pressing F5
- those requests only send back a few bytes (typically ~200 bytes) and an HHTP status = 304 (not modified) - which is expected
- however, since we're looking at dozends of CLR files this adds up and negatively impacts the user experience

Is there a way to improve the startup experience and e.g. bundle all the CLR files to one (or a few) before deploying to the 'web server' (which in my case is an AWS S3 bucket)? I browsed the docs and did some googling and saw that e.g. @MartinZikmund was experimenting with webpack, but did not find anything yet which would help.

Any suggestion/idea would be appreciated.

**Comment by MartinZikmund** [comment]
If I am not mistaken, this issue is not AOT specific but rather applies to All WASM modes, correct?

**Comment by DierkDroth** [comment]
@MartinZikmund hmm ... could be the case. I'm only 'living in full AOT land' at this point. Hence, I would not want to make statements on other setups...

**Reply by MartinZikmund** [reply]
I think support for this scenario needs to be added to the bootstrapper. Ideally it would also be helpful to allow "bundling" of the initial download of the libraries as a single request, because having a large number of requests is quite inefficient in case of slow cellular connections. Can you create an issue for this?

**Reply by DierkDroth** [reply]
[Done](https://github.com/unoplatform/uno/issues/13064)
